package/MDAnalysis/core/topology.py

```diff
@@ -122,8 +122,12 @@ def r2a_1d(self, rix):
             indices of atoms present in residues, collectively
 
         """
-        return np.concatenate([self._atom_order[x:y]
-                               for x, y in self._res_ptrs[rix]])
+        if isinstance(rix, int):
```

Comment:
Done. Moving to try-except blocks was possible for all downward moves except for the compound ones, since for those `numpy` arrays are iterable and wouldn't be caught.